File: apps/dash-oil-and-gas/points.py
# In[]:
# Import required libraries
import pandas as pd
import pickle


# In[]:
# Load required dataframes
dataset = dict()

columns1 = ['API Well Number', 'Gas Produced, MCF',
            'Water Produced, bbl', 'Reporting Year']
columns2 = ['API Well Number', 'Gas Produced, MCF',
            'Water Produced, bbl', 'Oil Produced, bbl', 'Reporting Year']
df1 = pd.read_csv('data/Oil_and_Gas_Annual_Production__1985_-_2000.csv',
                  low_memory=False)[columns1]
df2 = pd.read_csv('data/Oil_and_Gas_Annual_Production__Beginning_2001.csv',
                  low_memory=False)[columns2]


# In[]:
# Concatenate dataframes
df = pd.concat([df1, df2])
df.fillna(0, inplace=True)


# In[]:
# Append production data to each API
columns = ['Gas Produced, MCF', 'Water Produced, bbl',
           'Oil Produced, bbl', 'Reporting Year']
for api, df_well in df.groupby('API Well Number'):
    # print(api)
    # print(df_well)
    df_well = df_well[columns]
    df_well.index = df_well['Reporting Year']
    df_well = df_well.to_dict(orient='index')
    dataset[api] = df_well

# print(dataset)

# In[]:
# Serialise to Pickle
with open('data/points.pkl', 'wb') as f:
    pickle.dump(dataset, f)

File: apps/dashr-oil-and-gas/points.R
library(dplyr)
library(sqldf)
options(scipen=999)
library(data.table)

dataset <- list()
columns1 <- c('API.Well.Number', 'Gas.Produced..MCF',
            'Water.Produced..bbl', 'Reporting.Year')

columns2 <- c('API.Well.Number', 'Gas.Produced..MCF',
            'Water.Produced..bbl', 'Oil.Produced..bbl', 'Reporting.Year')

df1 = read.csv('data/Oil_and_Gas_Annual_Production__1985_-_2000.csv.gz')
df1 <- df1[,columns1]

df2 = read.csv('data/Oil_and_Gas_Annual_Production__Beginning_2001.csv.gz')
df2 <- df2[,columns2]

dfIn <- merge(df1,df2, all = TRUE)
dfIn[is.na(dfIn)] <- 0
dfIn <- dfIn[!duplicated(dfIn),]
colnames(dfIn)[1] <- 'API_WellNo'
dfIn <- as.data.table(dfIn)

columns = c('Gas.Produced..MCF', 'Water.Produced..bbl',
           'Oil.Produced..bbl', 'Reporting.Year')

dataset <- setNames(split(dfIn, seq(nrow(dfIn))), dfIn[,"API_WellNo"])
#dataset[[1]]['API.Well.Number']
#saveRDS(dataset, file="points.RData")






